worlds/metroidprime/data/ChozoRuins.py

```diff
@@ -0,0 +1,996 @@
+                ],
+            ),
+            RoomName.Dynamo_Access: RoomData(
+                include_area_in_name=True,
+                doors={
+                    0: DoorData(
+                        RoomName.Watery_Hall,
+                        blast_shield=BlastShieldType.Missile,
+                        rule_func=can_bomb,
+                    ),
+                    1: DoorData(
+                        RoomName.Dynamo, destination_area=MetroidPrimeArea.Chozo_Ruins
+                    ),
+                },
+            ),
+            RoomName.Dynamo: RoomData(
+                include_area_in_name=True,
+                doors={
+                    0: DoorData(
+                        RoomName.Dynamo_Access,
+                        destination_area=MetroidPrimeArea.Chozo_Ruins,
+                    ),
+                },
+                pickups=[
+                    PickupData("Chozo Ruins: Dynamo - Lower", rule_func=can_missile),
+                    PickupData(
+                        "Chozo Ruins: Dynamo - Spider Track",
+                        rule_func=lambda world, state: can_spider(world, state),
+                    ),
+                ],
+            ),
+            RoomName.East_Atrium: RoomData(
+                doors={
+                    0: DoorData(RoomName.Gathering_Hall),
+                    1: DoorData(RoomName.Energy_Core_Access),
+                }
+            ),
+            RoomName.East_Furnace_Access: RoomData(
+                doors={
+                    0: DoorData(
+                        RoomName.Hall_of_the_Elders, defaultLock=DoorLockType.Ice
+                    ),
+                    1: DoorData(RoomName.Furnace, defaultLock=DoorLockType.Ice),
+                }
+            ),
+            RoomName.Elder_Chamber: RoomData(
+                doors={
+                    0: DoorData(
+                        RoomName.Hall_of_the_Elders, defaultLock=DoorLockType.Ice
+                    ),
+                },
+                pickups=[PickupData("Chozo Ruins: Elder Chamber")],
+            ),
+            RoomName.Elder_Hall_Access: RoomData(
+                doors={
+                    0: DoorData(RoomName.Hall_of_the_Elders),
+                    1: DoorData(
+                        RoomName.Crossway, blast_shield=BlastShieldType.Missile
+                    ),
+                }
+            ),
+            RoomName.Energy_Core_Access: RoomData(
+                doors={
+                    0: DoorData(RoomName.Energy_Core),
+                    1: DoorData(RoomName.East_Atrium),
+                }
+            ),
+            RoomName.Energy_Core: RoomData(
+                doors={
+                    0: DoorData(
+                        RoomName.Burn_Dome_Access,
+                        defaultLock=DoorLockType.None_,
+                        rule_func=can_bomb,
+                        exclude_from_rando=True,
+                    ),  # Bombs are required to get out of here
```

Comment:
I'm putting this here because it's where the thought occurred to me, but this is more relevant to the game info document. If Point of No Return checks like this are never in logic, it's probably worth noting such in a player-facing manner to stave off some of the questions that may arise. If they are sometimes in logic, then this is inconsistent with that and I'm curious as to the discrepancy.

Reply:
Good call on this, I'll find somewhere to note this 🤔